dictgen/src/main.rs

```diff
@@ -32,9 +32,10 @@ struct Args {
     #[clap(long)]
     user_lexicon_out: Option<PathBuf>,
 
-    /// Outputs a list of features associated with each left connection ID.
+    /// Outputs a list of features associated with each connection ID and a list of bi-gram
+    /// weights.
```

Comment:
I guess the term `costs` is appropriate here rather than `weights` because inverse weights (i.e., `-1 * w`) are written in `write_bigram_details`.

Could you recheck the use of `weights` and `costs` in this update?